### ISSUE TITLE
Stacktrace output doesn't have clickable links

### ISSUE THREAD

**Issue by stinos** [JohnnyHendriks/TestAdapter_Catch2#32]
I'm not 100% sure this can be fixed, but in other test adapters the StackTrace output shown in the Test Explorer has clickable links which bring you to the failed test file/line, which is very handy.

The Catch2 adapter doesn't do that, so to get to the failed test one can go to the file (that one is clickable, first line in screenshot), but then has to manually go to the failed test line:

![Capture](https://user-images.githubusercontent.com/1902997/66909434-4a5e4580-effc-11e9-89ee-6a32d4092d44.PNG)

Hovering the lines in the stacktrace underlines them, but showing a tooltip with the full line and 'no source available'. So it looks like VS tries to parse this, but doesn't recognize it as a valid file/line combination, which possibly is because the StackTrace displayed doesn't adhere to the kind of format VS expects, i.e. maybe if the text would be printed differently it could be clicked.

Just for comparision, running the test executable as part of the build produces this output:

![Capture](https://user-images.githubusercontent.com/1902997/66909202-df147380-effb-11e9-81c5-b1bd0c3e2082.PNG)

in VS's Output Window and this has the correct format (`<file>(<line>)` AFAIK) to get parsed by VS which means the file/line combination can be doubleclicked to go to them.

**Comment by JohnnyHendriks** [comment]
Stacktrace links were enabled in version 1.2.0 of the TestAdapter for Catch2. That said, there are cases where it doesn't work, typically when there are line breaks in the stacktrace output. I actually already have changes in the pipeline that should make the stacktrace links more stable. So these should arrive with the next version of the TestAdapter for Catch2.

There are actually some examples in the [walkthrough](https://github.com/JohnnyHendriks/TestAdapter_Catch2/blob/master/Docs/Walkthrough.md#failed-test
) that show that the stacktrace links can work.

**Comment by stinos** [comment]
Yes sorry, my mistake. I didn't go into details at first and didn't realize it was only this particular example that doesn't work; if I change the value to check against 1 instead of `(size_t) -1` then Catch doesn't output line breaks and everything works nicely.

It's great if you;re working on this, but a nicer fix would probably be to tell Catch to not break lines in order to get deterministic output which is more suitable for parsing; I don't see runtime options for that, but thought that setting `CATCH_CONFIG_CONSOLE_WIDTH` at compile time would work. It doesn't, because the offending output gets broken in lines using a different value which cannot be configured. I'll fix that in local builds and raise an issue with Catch; looking at github there are some related issues (even getting xml output) but not this one exactly so I added https://github.com/catchorg/Catch2/issues/1775

Do I leave this open until you apply your stacktrace changes, or close because there's a workaround/open issue for Catch itself?

**Comment by JohnnyHendriks** [comment]
I have some control over the output from Catch2, in the sense that I use the xml reporter to retrieve information. So the stacktrace output is generated based on the information in the xml file. The solution now specifically checks for the presence of  line breaks. Also, I added an option to limit the length of the Stacktrace line.

The `CATCH_CONFIG_CONSOLE_WIDTH` should have no effect on the Test Adapter for Catch2, at least not for the test output. It does have some implications for the default test case discovery. Some tweaks are still needed there.

I have closed the issue as it has basically been fixed already, it is just not available in a released version yet. But I hope to release a new version within the next few weeks.
